### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by kinginthenorthcodez**
# Blog app - Views
- Implement the design from the sneak peek wireframes.
> Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using 
- Devise in coming milestone)
- All Users page wireframe Single User page wireframe All Posts by a given User page wireframe Single Post page wireframe

image
image

image

> Single User page wireframe
image

 > All Posts by a given User page wireframe
image
 > Single Post page wireframe
image

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- Hint: In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 <h3 style="color: #FF4433;">Pennode Users</h3>
 <p> Glad you joined us:</p>
-<p> Current Users: <ul style="color: orange;"> <% @users.each { |user| %> <li> <%= link_to user[:name], user_path(user) %> </li>  <% } %> </ul> </p>
\ No newline at end of file
+<p> Current Users: <ul style="color: orange;"> <% @users.each { |user| %> <li> <%= link_to user[:name], user_path(user)%> Number of Posts: <%= user[:posts_counter] %> </li>  <% } %> </ul> </p>
+<p>Here are a few of our fine users:</p>
+<% @users.each do |user| %>
+  <%= render partial:'shared/product', locals: { user: user }, spacer_template: 'shared/product_ruler' %>
+  
+<% end %>
```
Comment (FrederickMih) [inline app/views/users/index.html.erb]: - [ ] Hi @kinginthenorthcodez, once more let's make sure our index view page matches the wireframe below


Screenshot from 2022-08-26 18-47-15

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,13 +1,3 @@
...
- <td><%= @user[:photo] if @user %> </td>
- <td><%= @user[:posts_counter] if @user %> </td> 
- </tr>
-</table>
- </p>
-<p> Current User: <%= @user[:name] if @user %> </p>
-<h3> <%= link_to 'View posts',user_posts_path(@posts) if @posts %> </h3>
\ No newline at end of file
+<%= render partial: 'user', layout: 'shared/box', locals: { user: @user, posts: @current_user_posts } %>
+<h3> <%= link_to 'View posts',user_posts_path(@posts) if @posts %> </h3>
```
Comment (FrederickMih) [inline app/views/users/show.html.erb]: - [ ] Once more you'll again need to add some styles to the show page to at least match the wireframe below:

Screenshot from 2022-08-26 18-50-57

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <h3 style="color: #FF4433;">User Posts</h3>
+<%= render partial: 'post', layout: 'shared/box', locals: { user: @user, posts: @posts, comments: @post_comments, post_users: @post_users } %>
 <p> Glad you reading our posts:</p> 
 <p> Here is your Current Posts: <ul> <% @posts.each { |post| %> <li> <%= link_to post[:title],user_post_path(@user,post) %> </li>  <% } %> </ul> </p>
 <p> Some PARAM Data: <%= @values %></p>
```
Comment (FrederickMih) [inline app/views/posts/index.html.erb]: - [ ] Please let's modify the index page for all posts to look like the wireframe below:

Screenshot from 2022-08-26 19-53-19

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
+     <% if comment.post_id == @post.id && user.id == comment.user_id %>
+     <p style= "color: #900C3F; padding: 5px;">
+     <%= user.name %>  
+     <%= comment.text %> 
+     </p>
+     <% end %>
+      <% end %>
+    <% end %>
+  </div>
+
```
Comment (FrederickMih) [inline app/views/posts/show.html.erb]: - [ ] Lastly, kindly make sure that post show page looks like the wireframe below:

Screenshot from 2022-08-26 20-01-02

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @kinginthenorthcodez , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

- GitHub workflow has been followed :heavy_check_mark: 
- PR has a title and good summary :heavy_check_mark: 
- No Linter errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (kinginthenorthcodez) [reply app/views/users/index.html.erb]: @FrederickMih  Add suggested changes 👍🏽 
image

File: app/views/users/show.html.erb (hunk 2 above)
Comment (kinginthenorthcodez) [reply app/views/users/show.html.erb]: @FrederickMih  Added suggested changes 👍🏽 
image
image

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (kinginthenorthcodez) [reply app/views/posts/index.html.erb]: @FrederickMih  Added suggested Changes 👍🏽 
image
image

File: app/views/posts/show.html.erb (hunk 4 above)
Comment (kinginthenorthcodez) [reply app/views/posts/show.html.erb]: @FrederickMih  Added suggested changes 👍🏽 
image

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @kinginthenorthcodez, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights


- You have done a great job working on the requested changes✔️✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kinginthenorthcodez** [comment]
Merged #6  into development

**Pull request merged**